lib/plugin-loader.ts

```diff
@@ -0,0 +1,853 @@
+    Object.assign(this, this.def)
+  }
+
+  public validate() {
+    if (!semver.valid(this.version)) {
+      throw new Error(`ZWED0007E - ${this.name}: invalid version "${this.version}"`)
+    }
+    if (this.versionRequirements) {
+      for (let serviceName of Object.keys(this.versionRequirements)) {
+        if (!semver.validRange(this.versionRequirements[serviceName])) {
+          throw new Error(`ZWED0008E - ${this.localName}: invalid version range ` +
+          `${serviceName}: ${this.versionRequirements[serviceName]}`)
+        }
+      }
+    }
+  }
+}
+
+export class Import extends Service {
+  private versionRange: any;
+
+  constructor(def, configuration, plugin) {
+    super(def, configuration, plugin);
+  }
+
+  public validate() {
+    if (!semver.validRange(this.versionRange)) {
+      throw new Error(`ZWED0009E - ${this.localName}: invalid version range "${this.versionRange}"`)
+    }
+  }
+}
+
+export class NodeService extends Service {
+  private source: any;
+  private nodeModule: any;
+  private filename: any;
+  private fileName: any;
+
+  constructor(def, configuration, plugin) {
+    super(def, configuration, plugin);
+  }
+
+  public loadImplementation(dynamicallyCreated: boolean, location: string) {
+    if (dynamicallyCreated) {
+      const nodeModule = requireFromString(this.source);
+      this.nodeModule = nodeModule;
+    } else {
+      // Quick fix before MVD-947 is merged
+      var fileLocation = ""
+      var serverModuleLocation = ""
+      var clientModuleLocation = ""
+      if (this.filename) {
+        fileLocation = path.join(location, 'lib', this.filename);
+      } else if (this.fileName) {
+        fileLocation = path.join(location, 'lib', this.fileName);
+      } else {
+        throw new Error(`ZWED0010E - No file name for data service`)
+      
+      }
+      serverModuleLocation = path.join(location, 'nodeServer', 'node_modules' );
+      clientModuleLocation = path.join(location, 'lib', 'node_modules');
+      var nodePathStore = process.env.NODE_PATH;
+      var nodePathAdd = process.env.NODE_PATH;
+      var operatingSystem = process.platform;
+      var osDelim = operatingSystem === "win32" ? ";" : ":"; 
+      bootstrapLogger.info("ZWED0116I", serverModuleLocation, clientModuleLocation); //bootstrapLogger.log(bootstrapLogger.INFO,
+        //`The LOCATIONS are ${serverModuleLocation} and ${clientModuleLocation}`);
+      nodePathAdd = `${process.env.NODE_PATH}${serverModuleLocation}${osDelim}${clientModuleLocation}${osDelim}.`
+      process.env.NODE_PATH = nodePathAdd;
+      require("module").Module._initPaths();    
+      bootstrapLogger.info("ZWED0117I", fileLocation); //bootstrapLogger.log(bootstrapLogger.INFO, `The fileLocation is ${fileLocation}`);
+      bootstrapLogger.info("ZWED0118I", process.env.NODE_PATH); //bootstrapLogger.log(bootstrapLogger.INFO,
+        //`The NODE_PATH is ${process.env.NODE_PATH}`);
+      const nodeModule = require(fileLocation);
```

Comment:
require is often similar to typescript's `import`, except in this case. most require occur at the top of the file: a static require/import. I think import only works statically, while you can require dynamically by having require inside a function like we do here. But, typescript might give a compile warning, so you might want to do:
`const nodeModule = (require as any)(fileLocation);`
typecasting to `as any` will make annoying typescript messages disappear sometimes.